src/main/java/com/code_intelligence/jazzer/driver/FuzzTargetRunner.java

```diff
@@ -166,7 +170,21 @@ private static int runOne(long dataPtr, int dataLength) {
       //  1. Let libFuzzer pass in a null pointer if the byte array hasn't changed since the last
       //     call to our custom mutator and skip the read entirely.
       //  2. Implement a InputStream backed by Unsafe to avoid the copyToArray overhead.
-      mutator.read(new ByteArrayInputStream(copyToArray(dataPtr, dataLength)));
+      byte[] buf = copyToArray(dataPtr, dataLength);
+      boolean readCompletely = mutator.read(new ByteArrayInputStream(buf));
+
+      // Detect special libFuzzer inputs which can not be processed by the mutator framework.
+      boolean libFuzzerIndicator =
+          buf.length == 0 || (buf.length == 1 && buf[0] == LIBFUZZER_END_OF_CORPUS);
+
+      // If the mutator framework is used all constructed inputs are valid and read completely.
```

Comment:
I would find this easier to understand as "All inputs constructed by the mutator framework are valid and read completely.". 

What is "valid" referring to in this context? Aren't we using "read completely/exactly" as a proxy for that?

Reply:
Rephrased the comment a little, better?